tests/chmgt_test.go

```diff
@@ -0,0 +1,222 @@
+	response := executeRequest(req)
+
+	checkResponseCode(t, http.StatusOK, response.Code)
+
+	fmtresp, err := formatResponse(response)
+	if err != nil {
+		t.Errorf("Error in response format. Got %s", err)
+	}
+
+	if fmtresp.Success != true {
+		t.Errorf("Expected success to be 'true'. Got %v", fmtresp.Success)
+	}
+
+	if fmtresp.Message != "success" {
+		t.Errorf("Expected message to be 'success'. Got %v", fmtresp.Message)
+	}
+
+	body := formatData(fmtresp.Data)
+
+	if len(body) > 1 {
+		t.Errorf("Expected the body to only have 1 hit. Got %v", len(body))
+	}
+
+	if body[0]["username"] != "admin" {
+		t.Errorf("Expected the username to be 'admin'. Got %v", body[0]["username"])
+	}
+
+	if body[0]["email"] != "admin@example.com" {
+		t.Errorf("Expected the username to be 'admin@example.com'. Got %v", body[0]["email"])
+	}
+
+	if body[0]["firstname"] != "admin" {
+		t.Errorf("Expected the firstname to be 'Admin'. Got %v", body[0]["firstname"])
+	}
+
+	if body[0]["lastname"] != "User" {
+		t.Errorf("Expected the lastname to be 'User'. Got %v", body[0]["firstname"])
+	}
+}
+
+// TODO: Populate tests below
+
+// func TestGetMultipleUsers(t *testing.T) {
+
+// }
+
+// func TestGetNonExistingUser(t *testing.T) {
+
+// }
+
+// func TestAddUser(t *testing.T) {
+
+// }
+
+// func TestUpdateUser(t *testing.T) {
+
+// }
+
+// func TestDeleteUser(t *testing.T) {
+
+// }
+
+// func TestGetSingleChangeRequest(t *testing.T) {
+
+// }
+
+// func TestGetMultipleChangeRequest(t *testing.T) {
+
+// }
+
+// func TestGetNonExistingChangeRequest(t *testing.T) {
+
+// }
+
+// func TestAddChangeRequest(t *testing.T) {
+
+// }
+
+// func TestUpdateChangeRequest(t *testing.T) {
+
+// }
+
+// func TestDeleteChangeRequest(t *testing.T) {
+
+// }
+
+func executeRequest(req *http.Request) *httptest.ResponseRecorder {
+	rr := httptest.NewRecorder()
+	a.Router.ServeHTTP(rr, req)
+
+	return rr
+}
+
+func checkResponseCode(t *testing.T, expected, actual int) {
+	if expected != actual {
+		t.Errorf("Expected response code %d. Got %d\n", expected, actual)
+	}
+}
+
+func getAuthToken(user string, pw string) (string, error) {
+
+	authString := fmt.Sprintf(`{"username": "%s", "password": "%s"}`, user, pw)
+	data := []byte(authString)
+
+	req, err := http.NewRequest("POST", "/api/authenticate", bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	response := executeRequest(req)
+
+	resp1, err := formatResponse(response)
+	if err != nil {
+		return "", err
+	}
+
+	return resp1.Data.(string), nil
+}
+
+func formatResponse(r *httptest.ResponseRecorder) (handling.APIResponseJSON, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return handling.APIResponseJSON{}, err
+	}
+	response := handling.APIResponseJSON{}
+	jsonErr := json.Unmarshal(body, &response)
+
+	if jsonErr != nil {
+		return handling.APIResponseJSON{}, jsonErr
+	}
+
+	return response, nil
+}
```

Comment:
This might be a little cleaner:

```
func formatResponse(r *httptest.ResponseRecorder) (handling.APIResponseJSON, error) {
	response := handling.APIResponseJSON{}

	body, err := ioutil.ReadAll(r.Body)
	if err != nil {
		return response, err
	}

	err = json.Unmarshal(body, &response)
	return response, err
}
```